cms/static/coffee/spec/views/assets_spec.js

```diff
@@ -0,0 +1,433 @@
+                // return an error response
+                requests[0].respond(404);
+                expect(this.confirmationSpies.constructor).not.toHaveBeenCalled();
+                return expect(this.collection.contains(this.model)).toBeTruthy();
+            });
+
+            it("should lock the asset on confirmation", function() {
+                let requests;
+                ({view: this.view, requests} = this.createAssetView(this));
+
+                this.view.render().$(".lock-checkbox").click();
+                // AJAX request has been sent, but not yet returned
+                expect(this.model.save).toHaveBeenCalled();
+                expect(requests.length).toEqual(1);
+                expect(this.savingSpies.constructor).toHaveBeenCalled();
+                expect(this.savingSpies.show).toHaveBeenCalled();
+                const savingOptions = this.savingSpies.constructor.calls.mostRecent().args[0];
+                expect(savingOptions.title).toMatch("Saving");
+                expect(this.model.get("locked")).toBeFalsy();
+                // return a success response
+                requests[0].respond(204);
+                expect(this.savingSpies.hide).toHaveBeenCalled();
+                return expect(this.model.get("locked")).toBeTruthy();
+            });
+
+            return it("should not lock the asset if server errors", function() {
+                let requests;
+                ({view: this.view, requests} = this.createAssetView(this));
+
+                this.view.render().$(".lock-checkbox").click();
+                // return an error response
+                requests[0].respond(404);
+                // Don't call hide because that closes the notification showing the server error.
+                expect(this.savingSpies.hide).not.toHaveBeenCalled();
+                return expect(this.model.get("locked")).toBeFalsy();
+            });
+        });
+    });
+
+    return describe("Assets view", function() {
+        beforeEach(function(done) {
+            setFixtures($("<script>", {id: "asset-library-tpl", type: "text/template"}).text(assetLibraryTpl));
+            appendSetFixtures($("<script>", {id: "asset-tpl", type: "text/template"}).text(assetTpl));
+            window.analytics = jasmine.createSpyObj('analytics', ['track']);
+            window.course_location_analytics = jasmine.createSpy();
+            appendSetFixtures(sandbox({id: "asset_table_body"}));
+
+            this.promptSpies = jasmine.createSpyObj('Prompt.Warning', ["constructor", "show", "hide"]);
+            this.promptSpies.constructor.and.returnValue(this.promptSpies);
+            this.promptSpies.show.and.returnValue(this.promptSpies);
+
+            this.injector = new Squire();
+            this.injector.mock("common/js/components/views/feedback_prompt", {
+                "Warning": this.promptSpies.constructor
+            });
+
+            this.mockAsset1 = {
+                display_name: "test asset 1",
```

Comment:
This file is mixing single and double quotes, which again ESLint should complain about.